libobs/obs-output.c

```diff
@@ -2689,3 +2689,36 @@ const char *obs_output_get_supported_audio_codecs(const obs_output_t *output)
 		       ? output->info.encoded_audio_codecs
 		       : NULL;
 }
+
+const char *obs_output_get_protocols(const obs_output_t *output)
+{
+	if (!obs_output_valid(output, "obs_output_get_protocols"))
+		return NULL;
+
+	return (output->info.flags & OBS_OUTPUT_SERVICE)
+		       ? output->info.protocols
+		       : NULL;
+}
+
+bool obs_output_is_protocol_registered(const char *protocol)
+{
+	for (size_t i = 0; i < obs->output_types.num; i++) {
+		if (!(obs->output_types.array[i].flags & OBS_OUTPUT_SERVICE))
+			continue;
+
+		char **protocols = strlist_split(
+			obs->output_types.array[i].protocols, ';', false);
+
+		for (char **prtcl = protocols; *prtcl; ++prtcl) {
+			if (strcmp(*prtcl, protocol) != 0)
+				continue;
+
+			strlist_free(protocols);
+			return true;
+		}
+
+		strlist_free(protocols);
```

Comment:
More of a nitpick, but instead of splitting the string, allocating memory and all that, perhaps you'd want to consider simply looking at the string in slices (since all we care about is that there is a match)?

e.g.
```c
    const char *substr = obs->output_types.array[i].protocols;
    while (substr) {
        const char *next = strchr(substr, ';');
        size_t len = next ? (next - substr) : strlen(substr);
        if (strncmp(substr, protocol, len) == 0)
            return true;
        substr = next ? next + 1 : NULL;
    }
```

Reply:
This code might work but I'm not at ease at all with it.

Reply:
Yeah it's fine, just an optimisation. ´strtok()` could work as well and would only require one copy in this case. But unless this function gets called a lot, it doesn't matter to much.